src/server/detail/wrapped_json_path.h

```diff
@@ -14,16 +14,66 @@
 
 namespace dfly {
 
+class Nothing {};
 using JsonExpression = jsoncons::jsonpath::jsonpath_expression<JsonType>;
 
 template <typename T>
 using JsonPathEvaluateCallback = absl::FunctionRef<T(std::string_view, const JsonType&)>;
 
-template <typename T> class JsonCallbackResult {
+template <typename T = Nothing> class MutateCallbackResult {
  public:
+  MutateCallbackResult() = default;
+
+  explicit MutateCallbackResult(bool should_be_deleted_) : should_be_deleted(should_be_deleted_) {
+  }
```

Comment:
could've been just a freely accessible struct

Reply:
It does not work for case when `T` = `std::optional` and we want to save `std::nullopt `in it. `value_` will be `std::optional<std::optional<SomeClass>>` in this case. After returning `std::nullopt` from lambda` value_` will be empty, but we need to set second `std::optional` to `std::nullopt`

Reply:
that's a valid point, but in that case a constructor (bool, T value) would've been enough 🙂 Either way it's fine this way, just that is boilerplate code